manifests/coredns.yaml

```diff
@@ -66,13 +67,16 @@ data:
           reload 15s
           fallthrough
         }
+        import /etc/coredns/custom/*.override.fallthrough
         prometheus :9153
         forward . /etc/resolv.conf
         cache 30
         loop
         reload
         loadbalance
+        import /etc/coredns/custom/*.override.end
```

Comment:
This looks like it'll let you add things, but what if (for example) someone wanted to change the `forward .` line above. What does coredns do if they add a different `forward .` entry in an override.end, or in one of the other imported files up above?

Reply:
It would probably break things, as most of the CoreDNS plugins (including the `forward` plugin) are only allowed once per server block.
This case is particularly interesting though and I'd love to have a solution for it.
It would be amazing if there was something like a templating plugin for the Core file that would allow you to specify defaults that could be overridden by optional configs.

Reply:
Yeah, unfortunately what most people want would probably require switching over to a full Helm Chart for the coredns config so that we could use the HelmChartConfig support, which I am honestly not opposed to... It's just a much bigger change.

Reply:
I think the one main blocker for the customization in the current setup is that we have the `%{xyz}%` template variables in here that people don't want to miss. Could it be an option to e.g. check if a `coredns.yaml` exists on disk and if so, use that one (i.e. template it out) before using the one burned into the K3s binary? :thinking: 

But yeah, if you could give me a hint on where to start to add HelmChartConfig support, I'd love to try my hand on it :+1: 

For k3d specifically, what we have here would already be enough, as I could just add a new server block for a `.k3d.internal` zone that uses the `hosts` plugin but would be independent from the main block and the included `NodeHosts`.

Reply:
Lets strip it down to just the single extension point necessary for adding extra zones then; I don't want to create the false impression that these imports can be used to override existing entries in the corefile.

Reply:
Fine for me from the k3d point of view, but we'd have to check if it also solves the user need as pointed out in #462 and in the PR #743 :thinking: 
But I assume that e.g. the `rewrite` example could happen in an extra zone and the request would then start again in the zone of the rewritten name... so that would work :thinking: 
Let's go for it for now. Can/will the change be backported?

Reply:
Dropped the `import`s inside the main zone of the Corefile as per your suggestion @brandond (force-pushed to keep the commit history clean) :+1: